src/main/kotlin/com/pubnub/api/crypto/cryptor/AesCbcCryptor.kt

```diff
@@ -0,0 +1,83 @@
+package com.pubnub.api.crypto.cryptor
+
+import com.pubnub.api.PubNubError
+import com.pubnub.api.PubNubException
+import com.pubnub.api.crypto.data.EncryptedData
+import com.pubnub.api.crypto.data.EncryptedStreamData
+import java.io.InputStream
+import java.security.MessageDigest
+import java.security.SecureRandom
+import java.security.spec.AlgorithmParameterSpec
+import javax.crypto.Cipher
+import javax.crypto.spec.IvParameterSpec
+import javax.crypto.spec.SecretKeySpec
+
+private const val CIPHER_TRANSFORMATION = "AES/CBC/PKCS5Padding"
+
+class AesCbcCryptor(val cipherKey: String) : Cryptor {
+    private val keyBytes = sha256(cipherKey.toByteArray(Charsets.UTF_8))
+
+    override fun id(): ByteArray {
+        return byteArrayOf('A'.code.toByte(), 'C'.code.toByte(), 'R'.code.toByte(), 'H'.code.toByte())
+    }
+
+    override fun encrypt(data: ByteArray): EncryptedData {
+        return try {
+            val ivBytes: ByteArray = createRandomIv()
+            val ivSpec: AlgorithmParameterSpec = IvParameterSpec(ivBytes)
+            val newKey: SecretKeySpec = SecretKeySpec(keyBytes, "AES")
```

Comment:
I was actually thinking about taking out everything that can be taken out of those methods. That mean whole key creation can be done just once.

Reply:
ok,  I change this.